src/lib/comms/sol-lwm2m.c

```diff
@@ -2992,6 +2992,7 @@ handle_resource(struct sol_coap_server *server,
     r = extract_path(client, req, path, &path_size);
     header_code = SOL_COAP_RSPCODE_BAD_REQUEST;
     SOL_INT_CHECK_GOTO(r, < 0, exit);
+    SOL_INT_CHECK_GOTO(path_size, == 0, exit);
```

Comment:
I took another look and it seems that you are right, or at least I could not anymore reproduce the thought process I had when I was originally looking at the code. :-) This line appears to check that there is at least one item to process, which in turn meas that `path_size` can't be zero and `r` be non-negative at the same time:

https://github.com/ipuustin/soletta/blob/static-analysis-fixes/src/lib/comms/sol-lwm2m.c#L2261

I'll remove this patch from the PR...